### PR TITLE
Add suggestions for the test spec selector

### LINKED ISSUE #1383: Add suggestions for the test spec selector
**Context:** as part of the efforts of teaching users how to interact with our [Selector Language](https://docs.tracetest.io/advanced-selectors/), we want to provide a list of `suggestions` based on the currently selected span, that will help users understand how queries are built and illustrate common query operations.

**AC1:**
As a user interacting with the Create Test Spec form,
I want to see a list of `suggestions` for the selector query based on the selected span,
so I can easily apply and understand how to build queries.

**AC2:**
As a user interacting with the list of suggestions,
I want to click on one suggestion,
so I can apply the suggested query to the test spec.

**Proposed Rules:**
- All spans (`empty selector`)
- All spans by `type`
- All spans by `service`
- ~~All spans by `name`~~
- First span by `type`
- Last span by `type`
- All `type` spans that are children of parent span (only if span has a parent, it's not the root span)

### PR DESCRIPTION
**Pull request by jorgeepc**
This PR adds the logic to provide a list of `selector suggestions` based on the current selector (selected span), that will help users understand common query operations.

## Changes

- Backend: Add `selector structure` response to the `api/tests/${testId}/run/${runId}/select` endpoint.
- Frontend: Add `SelectorSuggestions` service. It implements and evaluate multiple `Rules` to provide suggestions based on different patterns.

## Fixes

- fixes #1383 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Loom

https://www.loom.com/share/d304f018be3542858b1765e4c365e90c

### CODE REVIEW

**Review by schoren — APPROVED** [review]
backend looks great!

**Review by mathnogueira — APPROVED** [review]
backend looks good

File: web/src/components/TestSpecForm/SelectorSuggestions.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    {Boolean(selectorSuggestions.length) && (
+      <>
+        <S.FormSectionText>
+          <InfoCircleOutlined /> Try to select spans using these suggestions:
+        </S.FormSectionText>
+
+        {Boolean(prevSelector) && (
+          <Tooltip title={prevSelector}>
+            <S.SuggestionsButton onClick={() => onClickPrevSelector(prevSelector)} type="link">
+              <ArrowLeftOutlined /> Prev selector
```
Comment (xoscar) [inline web/src/components/TestSpecForm/SelectorSuggestions.tsx:25]: I was looking into this in the UI and I'm not sure if we just need to have it more clear as an `undo` option

File: web/src/services/SelectorSuggestions/Rules.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import findLast from 'lodash/findLast';
+
+import {PseudoSelector} from 'constants/Operator.constants';
+import {TSelector} from 'types/Common.types';
+import {ISuggestion} from 'types/TestSpecs.types';
+import {ordinalSuffixOf} from 'utils/Common';
+
+interface IRule {
```
Comment (xoscar) [inline web/src/services/SelectorSuggestions/Rules.tsx:8]: Does this file need to have the `.tsx` extension?

File: web/src/services/SelectorSuggestions/SelectorSuggestions.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const rules = [
+  allSpansRule,
+  createByAttributeAllSpansRule(TRACETEST_SPAN_TYPE),
+  createByAttributeAllSpansRule(SERVICE_NAME),
+  structuralPseudoClassRule,
+  createByAttributeDescendantsRule(TRACETEST_SPAN_TYPE),
+  selectedSpanRule,
+];
+
+const SelectorSuggestionsService = {
```
Comment (xoscar) [inline web/src/services/SelectorSuggestions/SelectorSuggestions.service.ts:24]: Great job dude! really well done 🙌🏾

File: web/src/types/Common.types.ts [hunk 4]
```diff
@@ -41,13 +36,3 @@ export interface IAnalytics {
   track(event: string, traits: Record<string, any>): void;
   page(pageName: string, traits: Record<string, any>): void;
 }
-
```
Comment (xoscar) [inline web/src/types/Common.types.ts:44]: Great cleanup

File: web/src/utils/Common.ts [hunk 5]
```diff
@@ -34,11 +34,22 @@ export const downloadFile = (data: string, fileName: string): Element => {
...
-export function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
-  return Object.keys(obj).filter(k => Number.isNaN(Number(k))) as K[];
-}
-
 export function singularOrPlural(noun: string, quantity: number) {
   if (quantity === 1) return noun;
   return `${noun}s`;
 }
+
+export function ordinalSuffixOf(i: number) {
```
Comment (xoscar) [inline web/src/utils/Common.ts:42]: 🪄

**Review by xoscar — APPROVED** [review]
GJ @jorgeepc! 🔥  I left some nonblocking comments and suggestions

**Pull request merged**